http-client/src/client.rs

```diff
@@ -26,32 +30,33 @@ impl HttpClient {
 		let transport = HttpTransportClient::new(target, config).map_err(|e| Error::TransportError(Box::new(e)))?;
 		Ok(Self { transport, request_id: AtomicU64::new(0) })
 	}
+}
 
-	/// Send a notification to the server.
-	///
-	/// WARNING: This method must be executed on [Tokio 1.0](https://docs.rs/tokio/1.0.1/tokio).
-	pub async fn notification(
-		&self,
+impl Client for HttpClient {
+	type Error = Error;
+	type Subscription = ();
+
+	fn notification<'a>(
+		&'a self,
 		method: impl Into<String>,
 		params: impl Into<jsonrpc::Params>,
-	) -> Result<(), Error> {
+	) -> Pin<Box<dyn Future<Output = Result<(), Self::Error>> + Send + 'a>> {
```

Comment:
couldn't we avoid this with [`async-trait`](https://crates.io/crates/async-trait)

Reply:
yes :+1: